AttitudeManager/Src/AM.cpp

```diff
@@ -20,16 +20,18 @@ AttitudeManagerInput AttitudeManager::control_inputs = {
 };
 
 void AttitudeManager::setControlInputs(const AttitudeManagerInput& new_control_inputs) {
-    xSemaphoreTake( control_inputs_mutex, (TickType_t) 10);
-    control_inputs = new_control_inputs;
-    xSemaphoreGive( control_inputs_mutex );
+    if (xSemaphoreTake(control_inputs_mutex, (TickType_t) portMAX_DELAY) == pdPASS) {
+        control_inputs = new_control_inputs;
+        xSemaphoreGive(control_inputs_mutex);
+    }
 }
 
 AttitudeManagerInput AttitudeManager::getControlInputs() {
     struct AttitudeManagerInput temp;
```

Comment:
Since we return this if we fail to get the mutex, we should probably explicitly default initialize it. I'm also not sure we need to forward declare it.
```suggestion
    AttitudeManagerInput temp{};
```

Reply:
having temp is so we can run `xSemaphoreGive(control_inputs_mutex); ` before returning